profiler/CMakeLists.txt

```diff
@@ -87,6 +87,42 @@ if(THEROCK_ENABLE_ROCPROFV3)
   therock_cmake_subproject_provide_package(rocprofiler-sdk rocprofiler-sdk-roctx lib/cmake/rocprofiler-sdk-roctx)
   therock_cmake_subproject_activate(rocprofiler-sdk)
 
+  ##############################################################################
+  # rocprofiler-systems
+  ##############################################################################
+  therock_cmake_subproject_declare(rocprofiler-systems
+    USE_DIST_AMDGPU_TARGETS
+    EXTERNAL_SOURCE_DIR "${THEROCK_ROCM_SYSTEMS_SOURCE_DIR}/projects/rocprofiler-systems"
+    BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/rocprofiler-systems"
+    BACKGROUND_BUILD
+    CMAKE_ARGS
+      -DHIP_PLATFORM=amd
+      -DCMAKE_C_COMPILER=gcc
+      -DCMAKE_CXX_COMPILER=g++
```

Comment:
I think this may be our first subproject that tries to compile with gcc instead of either
1. the system toolchain (clang/MSVC)
2. amdclang that we build ourselves

That may be acceptable (cc @stellaraccident ), but I'd like to see some rationale written down in a comment here and maybe also in the PR description.

Did you try compiling with clang like so?
```cmake
    COMPILER_TOOLCHAIN
      amd-hip
```

Is there something unique about rocprofiler-systems that clang does not support but gcc does?

Reply:
We really can't be just ad-hoc hard-coding/naming a compiler here and hoping to get lucky.

Even if dyninst moves to third-party, this is adding gcc as a specific ROCM dependency. Can we discuss in office hours tomorrow? Doing this is going to be surprisingly fragile in various situations.

If we need to live with this situation for the short term, I'd rather have it build with the configured system compiler (which is the default), and you can add a check which specifically errors/warns if that is not GCC, giving a clear description and instructions for disabling rocprofiler-systems.